internal/provisioner/kustomize/kustomize.go

```diff
@@ -0,0 +1,179 @@
+package kustomize
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/yaml"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"github.com/operator-framework/rukpak/internal/util"
+)
+
+const (
+	// ProvisionerID is the unique kustomize provisioner ID
+	ProvisionerID = "core-rukpak-io-kustomize"
+)
+
+func HandleBundle(ctx context.Context, fsys fs.FS, bundle *rukpakv1alpha1.Bundle) (fs.FS, error) {
+	err := verifyObjects(".", fsys)
+	if err != nil {
+		return nil, err
+	}
+	return fsys, nil
+}
+
+func HandleBundleDeployment(ctx context.Context, fsys fs.FS, bd *rukpakv1alpha1.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
+	desiredObjects, err := loadBundle(fsys, bd)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	chrt := &chart.Chart{
+		Metadata: &chart.Metadata{},
+	}
+	for _, obj := range desiredObjects {
+		jsonData, err := yaml.Marshal(obj)
+		if err != nil {
+			return nil, nil, err
+		}
+		hash := sha256.Sum256(jsonData)
+		chrt.Templates = append(chrt.Templates, &chart.File{
+			Name: fmt.Sprintf("object-%x.yaml", hash[0:8]),
+			Data: jsonData,
+		})
+	}
+	return chrt, nil, nil
+}
+
+func verifyObjects(path string, bundleFS fs.FS) error {
+	entries, err := fs.ReadDir(bundleFS, path)
+	if err != nil {
+		return err
+	}
+	for _, e := range entries {
+		if e.IsDir() {
+			if (e.Type() & fs.ModeSymlink) != 0 {
```

Comment:
nit: Add a comment for explaining why we're filtering out this directory type?

Reply:
I'll add one.